integration_test/third_party_apps_test/main_test.go

```diff
@@ -173,7 +173,11 @@ func constructQuery(logName string, fields []*metadata.LogField) string {
 	var parts []string
 	for _, field := range fields {
 		if field.ValueRegex != "" {
-			parts = append(parts, fmt.Sprintf(`%s=~"%s"`, field.Name, field.ValueRegex))
+			if field.Optional {
+				parts = append(parts, fmt.Sprintf(`(NOT %s:* OR %s=~"%s")`, field.Name, field.Name, field.ValueRegex))
+			} else {
+				parts = append(parts, fmt.Sprintf(`%s=~"%s"`, field.Name, field.ValueRegex))
+			}
```

Comment:
Can we simplify:
```suggestion
			guard := ""
			if field.Optional {
				guard = fmt.Sprintf(`NOT %s:* OR `, field.Name)
			}
			parts = append(parts, fmt.Sprintf(`(%s%s=~"%s")`, guard, field.Name, field.ValueRegex))
```
?